### PR TITLE
ironbank enhancements to GitLab integration

### PR DESCRIPTION
**Pull request by guyfreund**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://github.com/demisto/etc/issues/40503

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by guyfreund** [comment]
Adding the `docs-approved` label as this is a community-supported pack.

File: Packs/DevSecOps/Integrations/GitLab/GitLab.py [hunk 1]
```diff
@@ -434,7 +697,14 @@ def main():
             'gitlab-pipelines-schedules-list': gitlab_pipelines_schedules_list_command,
             'gitlab-pipelines-list': gitlab_pipelines_list_command,
             'gitlab-jobs-list': gitlab_jobs_list_command,
-            'gitlab-artifact-get': gitlab_artifact_get_command
+            'gitlab-artifact-get': gitlab_artifact_get_command,
+            'gitlab-merge-requests-list': gitlab_merge_requests_list_command,
+            'gitlab-get-merge-request': gitlab_get_merge_request_command,
```
Comment (ShahafBenYakir) [inline Packs/DevSecOps/Integrations/GitLab/GitLab.py:702]:
```suggestion
            'gitlab-merge-request-get': gitlab_get_merge_request_command,
```

File: Packs/DevSecOps/Integrations/GitLab/GitLab.py [hunk 2]
```diff
@@ -434,7 +697,14 @@ def main():
...
+            'gitlab-issues-list': gitlab_issues_list_command,
+            'gitlab-create-issue': gitlab_create_issue_command,
```
Comment (ShahafBenYakir) [inline Packs/DevSecOps/Integrations/GitLab/GitLab.py:704]:
```suggestion
            'gitlab-issue-create': gitlab_create_issue_command,
```

File: Packs/DevSecOps/Integrations/GitLab/GitLab.py [hunk 3]
```diff
@@ -434,7 +697,14 @@ def main():
...
+            'gitlab-edit-issue': gitlab_edit_issue_command,
```
Comment (ShahafBenYakir) [inline Packs/DevSecOps/Integrations/GitLab/GitLab.py:705]:
```suggestion
            'gitlab-issue-edit': gitlab_edit_issue_command,
```

File: Packs/DevSecOps/Integrations/GitLab/GitLab.py [hunk 4]
```diff
@@ -434,7 +697,14 @@ def main():
...
+            'gitlab-group-projects-list': gitlab_group_projects_list_command,
+            'gitlab-get-raw-file': gitlab_get_raw_file_command
```
Comment (ShahafBenYakir) [inline Packs/DevSecOps/Integrations/GitLab/GitLab.py:707]:
```suggestion
            'gitlab-raw-file-get': gitlab_get_raw_file_command
```

File: Packs/DevSecOps/Integrations/GitLab/GitLab.yml [hunk 5]
```diff
@@ -641,7 +653,401 @@ script:
...
+      - contextPath: GitLab.MergeRequest.merge_commit_sha
+        description: The merge request merge commit SHA.
+        type: String
+      - contextPath: GitLab.MergeRequest.squash_commit_sha
+        description: The merge request squash commit SHA.
+        type: String
+  - arguments:
+    - name: labels
+      description: Comma seperated values of labels to search issues by.
+    - name: in
```
Comment (ShahafBenYakir) [inline Packs/DevSecOps/Integrations/GitLab/GitLab.yml:799]:
```suggestion
    - name: scope
```

Comment (guyfreund) [reply Packs/DevSecOps/Integrations/GitLab/GitLab.yml:799]: I think it is better to leave it as `in` as it is in the API.

Comment (ShahafBenYakir) [reply Packs/DevSecOps/Integrations/GitLab/GitLab.yml:799]: I guessed that this is the reason you chose this name but I don't think it is very clear. You can obviously set it to whatever you want later in the code.

**Review by ShahafBenYakir — APPROVED** [review]

**Pull request merged**